build.zig

```diff
@@ -301,3 +351,59 @@ fn buildShaders(b: *Build) void {
         shdc_step.dependOn(&cmd.step);
     }
 }
+
+fn buildWasm(b: *Build, example: *CompileStep, options: ExampleOptions) !*RunStep {
+    // system path or package path
+    const emcc_path = b.findProgram(&.{"emcc"}, &.{}) catch b.pathJoin(&.{ options.packagePath(b).?, "upstream", "emscripten", "emcc" });
+    const emrun_path = b.findProgram(&.{"emrun"}, &.{}) catch b.pathJoin(&.{ options.packagePath(b).?, "upstream", "emscripten", "emrun" });
+
+    if (options.lib_sokol.rootModuleTarget().os.tag != .emscripten) {
+        std.log.err("Please build with 'zig build -Dtarget=wasm32-emscripten", .{});
+        return error.Wasm32EmscriptenExpected;
+    }
+
+    // call the emcc linker step as a 'system command' zig build step which
+    // depends on the libsokol and libgame build steps
+    try std.fs.cwd().makePath(b.fmt("{s}/web", .{b.install_path}));
+
+    var emcc_cmds = std.ArrayList([]const u8).init(b.allocator);
+    defer emcc_cmds.deinit();
+
+    try emcc_cmds.append(emcc_path);
+    if (options.optimize != .Debug)
+        try emcc_cmds.append("-Oz")
+    else
+        try emcc_cmds.append("-Og");
+    try emcc_cmds.append("--closure");
+    try emcc_cmds.append("1");
+    try emcc_cmds.append(b.fmt("-o{s}/web/{s}.html", .{ b.install_path, example.name }));
+    try emcc_cmds.append("-sNO_FILESYSTEM=1");
+    try emcc_cmds.append("-sMALLOC='emmalloc'");
+    try emcc_cmds.append("-sASSERTIONS=0");
+    try emcc_cmds.append("-sERROR_ON_UNDEFINED_SYMBOLS=0");
+
+    // TODO: fix undefined references
+    // switch (options.backend) {
+    //     .wgpu => {
+    // try emcc_cmds.append("-sUSE_WEBGPU=1");
+    // },
+    // else => {
+    try emcc_cmds.append("-sUSE_WEBGL2=1");
+    //     },
+    // }
+    try emcc_cmds.append("-sEXPORTED_FUNCTIONS=['_malloc','_free','_main']");
```

Comment:
It looks like this `-sEXPORTED_FUNCTIONS` stuff is also no longer needed (not sure why tbh, but I just deleted it locally and the samples still build), and the clear sample still runs (and that definitely allocates).